src/property-filter/token-editor-grouped.tsx

```diff
@@ -0,0 +1,421 @@
+            <ValueInput
+              property={groups[index].property}
+              operator={groups[index].operator}
+              value={groups[index].value}
+              onChangeValue={groups[index].onChangeValue}
+              asyncProps={asyncProps}
+              filteringOptions={filteringOptions}
+              onLoadItems={onLoadItems}
+              i18nStrings={i18nStrings}
+            />
+          ),
+        }}
+      />
+
+      <div className={clsx(styles['token-editor-grouped-add-token'], testUtilStyles['token-editor-token-add-actions'])}>
+        <InternalButtonDropdown
+          variant="normal"
+          ariaLabel={i18nStringsExt.tokenEditorAddTokenActionsLabel}
+          items={standaloneTokens.map((token, index) => ({
+            id: index.toString(),
+            text: i18nStringsExt.tokenEditorAddExistingTokenLabel(getFormattedToken(token)),
+          }))}
+          onItemClick={({ detail }) => {
+            const index = parseInt(detail.id);
+            const addedToken = standaloneTokens[index];
+            const copy = [...standaloneTokens];
+            copy.splice(index, 1);
+            onChangeStandalone(copy);
+            onChangeTempGroup([...tempGroup, addedToken]);
+          }}
+          disabled={standaloneTokens.length === 0}
+          mainAction={{
+            text: i18nStringsExt.tokenEditorAddNewTokenLabel,
+            onClick: () => onChangeTempGroup([...tempGroup, { property: null, operator: ':', value: null }]),
+          }}
+        />
+      </div>
+
+      <div className={styles['token-editor-grouped-actions']}>
+        <InternalButton
+          formAction="none"
+          variant="link"
+          className={clsx(styles['token-editor-grouped-cancel'], testUtilStyles['token-editor-cancel'])}
+          onClick={onDismiss}
+        >
+          {i18nStrings.cancelActionText}
+        </InternalButton>
+        <InternalButton
+          className={clsx(styles['token-editor-grouped-submit'], testUtilStyles['token-editor-submit'])}
+          formAction="none"
+          onClick={onSubmit}
+        >
+          {i18nStrings.applyActionText}
+        </InternalButton>
+      </div>
+    </div>
+  );
+}
+
+interface TokenEditorLayout {
+  fields: number;
+  groupLabel: (index: number) => string;
+  removeButton: {
+    labelRemove: (index: number) => string;
+    labelRemoveFromGroup: (index: number) => string;
+    labelRemoveMore: (index: number) => string;
+    onRemove: (index: number) => void;
+    onRemoveFromGroup: (index: number) => void;
+  };
+  property: {
+    label: string;
+    render: (index: number) => React.ReactNode;
+  };
+  operator: {
+    label: string;
+    render: (index: number) => React.ReactNode;
+  };
+  value: {
+    label: string;
+    render: (index: number) => React.ReactNode;
+  };
+}
+
+function TokenEditorFields({ fields, groupLabel, removeButton, property, operator, value }: TokenEditorLayout) {
+  const breakpoint = 912;
```

Comment:
Where is 912 coming from? Should we export this as a constant somewhere?

Reply:
yes, we have a utility we can use - will update